pallets/token-dealer/src/support.rs

```diff
@@ -0,0 +1,256 @@
+// Copyright 2019-2020 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! XCM objects and relationships
+use frame_support::{
+	debug,
+	traits::{Currency, ExistenceRequirement, Get, WithdrawReasons},
+};
+use parity_scale_codec::{Decode, Encode};
+use sp_runtime::{
+	traits::{CheckedConversion, Convert},
+	RuntimeDebug,
+};
+use sp_std::{
+	collections::btree_set::BTreeSet,
+	convert::{TryFrom, TryInto},
+	marker::PhantomData,
+	prelude::*,
+	result,
+};
+use xcm::v0::{Error, Junction, MultiAsset, MultiLocation, Result as XcmResult};
+use xcm_executor::traits::{
+	FilterAssetLocation, LocationConversion, MatchesFungible, NativeAsset, TransactAsset,
+};
+
+#[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug, PartialOrd, Ord)]
+#[non_exhaustive]
+pub enum TokenSymbol {
```

Comment:
My first approach was `TokenId = Vec<u8>` but we need a type that can be a constant variable (for runtime parameter `RelayChainCurrencyId`) and you can't declare constant vectors (because constant variables have to be defined at compile time and vectors are heap allocated at runtime).

This enum adds some friction though because it makes the flow for adding a new ERC token to token-factory:
1. add variant to this enum for the new token i.e. `TokenSymbol::Test`
2. call register dispatchable in `token_factory` with the new variant i.e. `register_token(TokenSymbol::Test)`
I wonder if we could write client code that does both in one step. The enum is annotated with `non_exhaustive` with the expectation that more variants will be added.